src/editor/codemirror/dropcursor.ts

```diff
@@ -0,0 +1,166 @@
+}
+
+const setDropCursorPos = StateEffect.define<number | null>({
+  map(pos, mapping) {
+    return pos == null ? null : mapping.mapPos(pos);
+  },
+});
+
+const dropCursorPos = StateField.define<number | null>({
+  create() {
+    return null;
+  },
+  update(pos, tr) {
+    if (pos != null) pos = tr.changes.mapPos(pos);
+    return tr.effects.reduce(
+      (pos, e) => (e.is(setDropCursorPos) ? e.value : pos),
+      pos
+    );
+  },
+});
+
+const drawDropCursor = ViewPlugin.fromClass(
+  class {
+    cursor: HTMLElement | null = null;
+    measureReq: MeasureRequest<{
+      left: number;
+      top: number;
+      height: number;
+    } | null>;
+
+    constructor(readonly view: EditorView) {
+      this.measureReq = {
+        read: this.readPos.bind(this),
+        write: this.drawCursor.bind(this),
+      };
+    }
+
+    update(update: ViewUpdate) {
+      let cursorPos = update.state.field(dropCursorPos);
+      if (cursorPos == null) {
+        if (this.cursor != null) {
+          this.cursor?.remove();
+          this.cursor = null;
+        }
+      } else {
+        if (!this.cursor) {
+          this.cursor = this.view.scrollDOM.appendChild(
+            document.createElement("div")
+          );
+          this.cursor!.className = "cm-dropCursor";
+        }
+        if (
+          update.startState.field(dropCursorPos) != cursorPos ||
+          update.docChanged ||
+          update.geometryChanged
+        )
+          this.view.requestMeasure(this.measureReq);
+      }
+    }
+
+    readPos(): { left: number; top: number; height: number } | null {
+      let pos = this.view.state.field(dropCursorPos);
+      let rect = pos != null && this.view.coordsAtPos(pos);
+      if (!rect) return null;
+      let outer = this.view.scrollDOM.getBoundingClientRect();
+      return {
+        left: rect.left - outer.left + this.view.scrollDOM.scrollLeft,
+        top: rect.top - outer.top + this.view.scrollDOM.scrollTop,
+        height: rect.bottom - rect.top,
+      };
+    }
+
+    drawCursor(pos: { left: number; top: number; height: number } | null) {
+      if (this.cursor) {
+        if (pos) {
+          this.cursor.style.left = pos.left + "px";
+          this.cursor.style.top = pos.top + "px";
+          this.cursor.style.height = pos.height + "px";
+        } else {
+          this.cursor.style.left = "-100000px";
+        }
+      }
+    }
+
+    destroy() {
+      if (this.cursor) this.cursor.remove();
+    }
+
+    setDropPos(pos: number | null) {
+      if (this.view.state.field(dropCursorPos) != pos)
+        this.view.dispatch({ effects: setDropCursorPos.of(pos) });
+    }
+  },
+  {
+    eventHandlers: {
+      dragover(event) {
+        // Avoid setting drop position if dragged element is a code snippet/example.
+        if (!(event.target as HTMLElement).classList.contains("cm-content")) {
+          this.setDropPos(
+            this.view.posAtCoords({ x: event.clientX, y: event.clientY })
+          );
+        }
```

Comment:
@microbit-matt-hillsdon This works but feels like it could be flimsy. Open to other suggestions to determine what is being dragged.

Reply:
One of event.dataTransfer.types is equal to our `pythonSnippetMediaType` constant?

Reply:
Perfect - [`94eaf98`](https://github.com/microbit-foundation/python-editor-next/pull/613/commits/94eaf98b0bd76e07c988cbc40e97f279fc39db09)